libraries/api/test/validate_test.js

```diff
@@ -296,4 +332,40 @@ suite(testing.suiteName(), function() {
     }
     assert(0, 'Did not get expected exception');
   });
+  test('calling send twice with status json triggers an error', () => {
+    const url = u('/test-double-send');
+    return request.get(url).then(function(res) {
+      assert(false, 'the error didnt trigger');
+    }).catch(function (err) {
+      assert(true, 'Error did trigger');
+      assert(helper.monitorManager.messages[0].Fields.message === 'called send twice');
+    });
+  });
+
+  test('calling send twice with reportError triggers an Error', () => {
+
+    const url = u('/test-double-error-send');
+    return request.get(url).then((res) => {
+      assert(false, 'Error didn\'t trigger');
+    }).catch(() => {
+      assert(true, 'Error was triggered');
+      setTimeout( () => {
+        assert(helper.monitorManager.messages[0].Fields.message === 'called send twice');
+      }, 100);
```

Comment:
This is a subtle bug!  The test here is synchronous, so as soon as the test function returns, Mocha marks it as passed and starts the next test. Then 100ms later, this setTimeout callback fires, and checks that the message exists because of the *next* test, not this one!

See the note above as to why this test case doesn't fail without the setTimeout.